GitVersionCore/BranchNameComparer.cs

```diff
@@ -0,0 +1,18 @@
+namespace GitVersion
+{
+    using System.Collections.Generic;
+    using LibGit2Sharp;
+
+    class BranchNameComparer : IEqualityComparer<Branch>
+    {
+        public bool Equals(Branch x, Branch y)
+        {
+            return x.Name == y.Name;
```

Comment:
`Branch`  supports `Equals()`, `==`, `!=` and `GetHashCode()`. The `CanonicalName` and the `Id` of the pointed at `GitObject` will be used to assert the equality. `CanonicalName`s will be compared in a case sensitive way. Those things matter on non case insensitive file systems.

Although if you decide to name two branches `master` and `mAsTeR`, you may deserve to encounter some problems :wink: